msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/AcquireTokenSilentlyTest.java

```diff
@@ -116,6 +119,79 @@ void confidentialAppAcquireTokenSilently_claimsSkipCache() throws Throwable {
         assertInstanceOf(MsalInteractionRequiredException.class, ex.getCause());
     }
 
+    @Test
+    void testTokenRefreshReasons() throws Exception {
+        DefaultHttpClient httpClientMock = mock(DefaultHttpClient.class);
+
+        ConfidentialClientApplication cca =
+                ConfidentialClientApplication.builder("clientId", ClientCredentialFactory.createFromSecret("password"))
+                        .authority("https://login.microsoftonline.com/tenant/")
+                        .instanceDiscovery(false)
```

Comment:
nit: won't block the PR for this, but in general you do not want to over-customize MSAL. The Http mocking can reply with instance discovery json.

Reply:
Since these unit tests weren't covering anything related to instance discovery I figured it'd be easier to just disable these extra HTTP calls, but I agree that's bad practice. I'll create an issue to improve the HTTP mocking since it is such a core part of the unit tests.